cmake/CMakeLists.txt

```diff
@@ -290,25 +290,8 @@ if(BUILD_MPI)
     endif()
   endif()
 else()
-  file(GLOB MPI_SOURCES ${LAMMPS_SOURCE_DIR}/STUBS/mpi.cpp)
-  add_library(mpi_stubs STATIC ${MPI_SOURCES})
-  set_target_properties(mpi_stubs PROPERTIES OUTPUT_NAME lammps_mpi_stubs${LAMMPS_MACHINE})
-  target_include_directories(mpi_stubs PUBLIC $<BUILD_INTERFACE:${LAMMPS_SOURCE_DIR}/STUBS>)
-  if(BUILD_SHARED_LIBS)
-    target_link_libraries(lammps PRIVATE mpi_stubs)
-    if(MSVC)
-      target_link_libraries(lmp PRIVATE mpi_stubs)
-      target_include_directories(lmp INTERFACE $<BUILD_INTERFACE:${LAMMPS_SOURCE_DIR}/STUBS>)
-      target_compile_definitions(lmp INTERFACE $<INSTALL_INTERFACE:LAMMPS_LIB_NO_MPI>)
-    endif()
-    target_include_directories(lammps INTERFACE $<BUILD_INTERFACE:${LAMMPS_SOURCE_DIR}/STUBS>)
-    target_compile_definitions(lammps INTERFACE $<INSTALL_INTERFACE:LAMMPS_LIB_NO_MPI>)
-  else()
-    target_include_directories(lammps INTERFACE $<BUILD_INTERFACE:${LAMMPS_SOURCE_DIR}/STUBS>)
-    target_compile_definitions(lammps INTERFACE $<INSTALL_INTERFACE:LAMMPS_LIB_NO_MPI>)
-    target_link_libraries(lammps PUBLIC mpi_stubs)
-  endif()
-  add_library(MPI::MPI_CXX ALIAS mpi_stubs)
+  target_sources(lammps PRIVATE ${LAMMPS_SOURCE_DIR}/STUBS/mpi.cpp)
+  target_include_directories(lammps PUBLIC ${LAMMPS_SOURCE_DIR}/STUBS)
```

Comment:
```suggestion
  target_include_directories(lammps PUBLIC $<BUILD_INTERFACE:${LAMMPS_SOURCE_DIR}/STUBS>)
```

Reply:
Made the change because I trust you, but please explain why is this important to a non-guru person.

Reply:
PUBLIC include directories will be added to the lammps exported target during install. But having a path from the source directory there makes not much sense as the source might be deleted after install.

Reply:
Until we have made changes to the LAMMPS source tree to have a well defined C++ API for the library, use a version specific SONAME, and have a lammpsconfig.h header that records all defines instead of depending on using some method of passing required flags, the "make install" functionality should be considered "unsupported". I understand that this is useful for what you do (or want to do), but from my perspective this is subject to review and refactoring/redesign at a later point. Thus these kinds of adjustments and considerations are not going to be of any priority. It was easy enough in this particular case, so it was no problem this time.